commvault/operations.py

```diff
@@ -0,0 +1,138 @@
+""" Copyright start
+  Copyright (C) 2008 - 2022 Fortinet Inc.
+  All rights reserved.
+  FORTINET CONFIDENTIAL & FORTINET PROPRIETARY SOURCE CODE
+  Copyright end """
+
+import base64
+from requests import request, exceptions as req_exceptions
+from connectors.core.connector import get_logger, ConnectorError
+from connectors.core.utils import update_connnector_config
+from .constant import *
+from datetime import datetime, timedelta
+
+
+logger = get_logger("commvault")
+
+
+class Commvault:
+    def __init__(self, config, connector_info, *args, **kwargs):
+        """
+        For documentation refer
+        1. https://documentation.commvault.com/v11/essential/45578_rest_api_authentication_post_login.html
+        2. https://documentation.commvault.com/v11/essential/45551_rest_api_postman.html
+        Note: The token by default is valid for 30 minutes
+        """
+        pwd = bytes(config.get("password"), encoding='utf8')
+        pwd = str(base64.b64encode(pwd), encoding='utf-8')
+        server_url = config.get("server_url")
+        if not server_url.startswith('https://') and not server_url.startswith('http://'):
+            server_url = "https://"+server_url
+        self.url = server_url
+        self.username = config.get("username")
+        self.password = pwd
+        self.verify_ssl = config.get("verify_ssl")
+        self.token = self.get_token(config, connector_info, **kwargs)
+
+    def get_token(self, config, connector_info, **kwargs):
+        token = config.get("token")
+        expiry_in = config.get("expiry_in")
+        check_health = kwargs.get("check_health")
+        if not check_health and token and expiry_in and datetime.fromtimestamp(int(expiry_in)) > datetime.now():
+            return token
+        else:
+            data = {
+              "username": self.username,
+              "password": self.password
+            }
+            current_time = datetime.now() + timedelta(minutes=EXPIRY_TIME)
+            headers = dict()
+            headers["Accept"] = "application/json"
+            headers["Content-Type"] = "application/json"
+            response = self.api_request(Method.POST, Endpoint.LOGIN, self.url+Endpoint.LOGIN, data=data)
+            token = response.get("token")
+            config["token"] = token
+            config["expiry_in"] = int(current_time.timestamp())
+            update_connnector_config(connector_info["connector_name"],
+                                     connector_info["connector_version"],
+                                     config,
+                                     config["config_id"])
+            return token
+
+    def api_request(self, method, endpoint, params={}, data={}):
```

Comment:
can we pass 'GET' as the default value for the method parameter since most of the actions are of get type

Reply:
optional arguments are given default values, if we want to hit request, method and endpoint are required so this should be kept as compulsory arguments